doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,177 @@
+..   list and it is being improved by its author.  The mailing list
+..   discussion of the MEP should include the MEP number (MEPxxx) in the
+..   subject line so they can be easily related to the MEP.
+
+.. - **Progress**: Consensus was reached on the mailing list and
+..   implementation work has begun.
+
+.. - **Completed**: The implementation has been merged into master.
+
+.. - **Superseded**: This MEP has been abandoned in favor of another
+     approach.
+
+Branches and Pull requests
+==========================
+
+* `Issue 1109 <https://github.com/matplotlib/matplotlib/issues/1109>`_
+* `Initial thread (mailing list) <https://www.mail-archive.com/matplotlib-devel%40lists.sourceforge.net/msg11325.html>`_
+* `MEP28 PR <https://github.com/matplotlib/matplotlib/pull/4384>`_
+
+
+Abstract
+========
+
+This MEP proposes a new graphical way of specifying simple to moderately
+complex subplot layouts using arrays of strings.
+
+
+Detailed description
+====================
+
+The most popular way of specifying subplots layout is to use the ``subplot``
+function. This ``subplot`` function is rather unintuitive and makes it
+difficult to specify moderately complex subplots. There are alternatives, some
+of them very powerful, but they generally require a lot of code to specifiy
+precisely the layout. There might be room for an in-between method that rely on
+the graphical description of the layout. It can cover all ``subplot`` cases and
+should allow to specify moderately complex subplots.
+
+Examples
+--------
+
+::
+
+  "AB"
+  ┌────────┐┌────────┐
+  │ A      ││ B      │
+  │        ││        │
+  │        ││        │
+  └────────┘└────────┘
+
+  "ABB"
+  ┌──────┐┌──────────┐
+  │ A    ││ B        │
+  │      ││          │
+  │      ││          │
+  └──────┘└──────────┘
+
+  "ABD"
+  "CCD"
+  ┌───────┐┌───────┐┌───────┐
+  │ A     ││ B     ││ D     │
+  │       ││       ││       │
+  │       ││       ││       │
+  └───────┘└───────┘│       │
+  ┌────────────────┐│       │
+  │ C              ││       │
+  │                ││       │
+  │                ││       │
+  └────────────────┘└───────┘
+
+  "AaBb"
+  ┌───────┐┌─┐┌───────┐┌─┐
+  │ A     ││ ││ B     ││ │
+  │       ││ ││       ││ │
+  │       ││ ││       ││ │
+  └───────┘└─┘└───────┘└─┘
+
+  "  b  "
+  "aABCc"
+              ┌───────┐
+              └───────┘
+  ┌─┐┌───────┐┌───────┐┌───────┐┌─┐
+  │ ││ A     ││ B     ││ C     ││ │
+  │ ││       ││       ││       ││ │
+  │ ││       ││       ││       ││ │
+  └─┘└───────┘└───────┘└───────┘└─┘
+
+
+Syntax
+------
+
+A subplot is described using a capital letter, from ``A`` to ``Z`` such that
+only 26 subplots can be specified at once. Colorbars are specified using
+non-capital letters, from ``a`` to ``z``. The main command is the ``layout``
+command that returns a list of axes (subplots and colorbars) following the
+lexicographic order, mixing capital and small letters. For example::
+
+  A,B,C = layout("ABC")
+  A,B,b = layout("ABb")
```

Comment:
I'm not crazy about the lexicographic ordering requirement here (and it also makes the Unicode case pointed out by @OceanWolf more difficult).

What if instead of a tuple, it returns a dictionary, i.e.:

```
>>> layout("ABC")
{"A": <Axes>, "B": <Axes>, "C": <Axes>}
```

We could even have things both ways by returning an OrderedDict ordered as you suggest here.  (That way you get the concision of tuple unpacking for the simple case where the ordering isn't so complicated).